### ISSUE TITLE
0.6.6 test suite fails with Python 2.4.3 on Rhel 5

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#440]
Hello there,

first of all many thanks for Sphinx.  I have been putting together an up-to-date Sphinx package for my team on Rhel5 and get the errors below when I run the test suite.  The 0.6.5 test suite runs without any problems on the same platform. An up-to-date Fedora 12 also runs the 0.6.6 test without problems.

The Rhel5 system has the following Python stack to support Sphinx.

python-2.4.3-27.el5.x86_64
python-nose-0.11.3-4.hp.el5.noarch
python-jinja2-2.5-1.hp.el5.x86_64
python-babel-0.9.5-2.hp.el5.noarch
python-docutils-0.6-2.hp.el5.noarch
python-imaging-1.1.5-5.el5.x86_64

All the best,

Gareth

```
[gareth@repoman Sphinx-0.6.6]$ nosetests 
.........E.F.................
======================================================================
ERROR: test_env.test_first_update
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.4/site-packages/nose/case.py", line 182, in runTest
    self.test(*self.arg)
  File "/root/Sphinx-0.6.6/tests/test_env.py", line 43, in test_first_update
    for docname in it:  # the generator does all the work
  File "/root/Sphinx-0.6.6/sphinx/environment.py", line 513, in update_generator
    self.read_doc(docname, app=app)
  File "/root/Sphinx-0.6.6/sphinx/environment.py", line 604, in read_doc
    pub.publish()
  File "/usr/lib/python2.4/site-packages/docutils/core.py", line 204, in publish
    self.settings)
  File "/usr/lib/python2.4/site-packages/docutils/readers/__init__.py", line 69, in read
    self.parse()
  File "/usr/lib/python2.4/site-packages/docutils/readers/__init__.py", line 75, in parse
    self.parser.parse(self.input, document)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/__init__.py", line 157, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 170, in run
    input_source=document['source'])
  File "/usr/lib/python2.4/site-packages/docutils/statemachine.py", line 231, in run
    context, next_state, result = self.check_line(
  File "/usr/lib/python2.4/site-packages/docutils/statemachine.py", line 420, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 2658, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 308, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 376, in new_subsection
    node=section_node, match_titles=1)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 266, in nested_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python2.4/site-packages/docutils/statemachine.py", line 231, in run
    context, next_state, result = self.check_line(
  File "/usr/lib/python2.4/site-packages/docutils/statemachine.py", line 420, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 2241, in explicit_markup
    self.explicit_list(blank_finish)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 2269, in explicit_list
    match_titles=self.state_machine.match_titles)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 300, in nested_list_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python2.4/site-packages/docutils/statemachine.py", line 231, in run
    context, next_state, result = self.check_line(
  File "/usr/lib/python2.4/site-packages/docutils/statemachine.py", line 420, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 2542, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 2251, in explicit_construct
    return method(self, expmatch)
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 1993, in directive
    return self.run_directive(
  File "/usr/lib/python2.4/site-packages/docutils/parsers/rst/states.py", line 2043, in run_directive
    result = directive_instance.run()
  File "/root/Sphinx-0.6.6/sphinx/directives/code.py", line 121, in run
    codec_info.streamreader, codec_info.streamwriter, 'strict')
AttributeError: 'tuple' object has no attribute 'streamreader'

======================================================================
FAIL: test_env.test_second_update
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.4/site-packages/nose/case.py", line 182, in runTest
    self.test(*self.arg)
  File "/root/Sphinx-0.6.6/tests/test_env.py", line 83, in test_second_update
    assert '1 added, 3 changed, 1 removed' in msg
AssertionError

----------------------------------------------------------------------
Ran 29 tests in 3.285s

FAILED (errors=1, failures=1)
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/440
- Originally reported by: [g_armstrong](https://bitbucket.org/g_armstrong)
- Originally created at: 2010-06-01T15:43:22.619

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-06-05 16:42:26+00:00_

Thanks for the report.  This was indeed a change in 0.6.6 that uses a Python >= 2.5 API in codecs.py.  I've fixed that in the 0.6 branch and will release a 0.6.7 with the fix today.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-11-01 16:54:57+00:00_

fix #440 #1008 : coarse timestamp resolution in some filesystem generate wrong outdated file-list

→ <<cset 718993afc8e0e61c689a7681aa27f0f6dfbbcb7e>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-03-31 13:57:31+00:00_

Removing version: 0.6.6 (automated comment)
